src/poetry/puzzle/provider.py

```diff
@@ -670,10 +670,9 @@ def complete_package(
             #   - {<Package foo (1.2.3): {"bar": <Dependency bar (<2.0)>}
 
             def fmt_warning(d: Dependency) -> str:
-                marker = d.marker if not d.marker.is_any() else "*"
                 return (
                     f"<c1>{d.name}</c1> <fg=default>(<c2>{d.pretty_constraint}</c2>)</>"
-                    f" with markers <b>{marker}</b>"
+                    f" with markers <b>{'*' if d.marker.is_any() else d.marker}</b>"
```

Comment:
While this change is not strictly needed, it raised a `B023` as well.

As `marker` is also defined outside of `fmt_warning`, I believe it confused the plugin (because renaming `marker` to `dependency_marker` made it pass).

Since the variable is only used once, I though it wasn't worth having one anyway, but if we prefer the current version, it's fine by me to just rename `marker` to `dependency_marker` instead.

Reply:
I think it's good the way it is

Reply:
Alright, squashed into https://github.com/python-poetry/poetry/pull/6095/commits/a84feef0fd66ec25f3c0f4ef9a5c738d39bbf83a by only renaming `marker` to `dependency_marker`.